geps/gep-2895/index.md

```diff
@@ -0,0 +1,265 @@
+    // parameter from an HTTP request before it is sent to the upstream target.
+    //
+    // Support in HTTPRouteRule: Extended
+    //
+    // Support in HTTPBackendRef: Extended
+    HTTPRouteFilterQueryParamModifier HTTPRouteFilterType = "QueryParamModifier"
+)
+
+// HTTPQueryParamFilter defines a filter that modifies HTTP query parameter.
+// Only one action for a given query param name is permitted.
+// Filters specifying multiple actions of the same or different type for any one
+// query param name are invalid and will be rejected by CRD validation.
+type HTTPQueryParamFilter struct {
+    // Set overwrites the HTTP request with the given query param (name, value)
+    // before the action. 
+    // The request query parameter names are case-sensitive.
+    // This must be an exact string match of query param name.
+    // (See https://www.rfc-editor.org/rfc/rfc7230#section-2.7.3).
+    //
+    // Input:
+    //   GET /foo?my-parameter=foo HTTP/1.1
+    //
+    // Config:
+    //   set:
+    //   - name: "my-parameter"
+    //     value: "bar"
+    //
+    // Output:
+    //   GET /foo?my-parameter=bar HTTP/1.1
+    //
+    // If the query parameter is not set in the request,
+    // the Set action MUST be ignored by the Gateway.
+    //
+    // Input:
+    //   GET /foo HTTP/1.1
+    //
+    // Config:
+    //   set:
+    //   - name: "my-parameter"
+    //     value: "bar"
+    //
+    // Output:
+    //   GET /foo HTTP/1.1
+    //
+    // +optional
+    // +listType=map
+    // +listMapKey=name
+    // +kubebuilder:validation:MaxItems=16
+    Set []HTTPHeader `json:"set,omitempty"`
+
+    // Add adds the given query param(s) (name, value) to the HTTP request
+    // before the action. Existing query params with the same name are not 
+    // replaced, instead a new param with the same name is added.
+    //
+    // Input:
+    //   GET /foo?my-parameter=foo HTTP/1.1
+    //
+    // Config:
+    //   add:
+    //   - name: "my-parameter"
+    //     value: "bar"
+    //
+    // Output:
+    //   GET /foo?my-parameter=foo&my-parameter=bar HTTP/1.1
+    //
+    // +optional
+    // +listType=map
+    // +listMapKey=name
+    // +kubebuilder:validation:MaxItems=16
+    Add []HTTPHeader `json:"add,omitempty"`
+
+    // Remove the given query param(s) from the HTTP request before the action.
+    // The value of Remove is a list of query param names. Note that the query
+    // param names are case-sensitive (See
+    // https://www.rfc-editor.org/rfc/rfc7230#section-2.7.3).
+    //
+    // Input:
+    //   GET /foo?my-parameter1=foo&my-parameter2=bar&my-parameter3=baz HTTP/1.1
+    //
+    // Config:
+    //   remove: ["my-parameter1", "my-parameter3"]
+    //
+    // Output:
+    //   GET /foo?my-parameter2=bar HTTP/1.1
+    //
+    // +optional
+    // +listType=set
+    // +kubebuilder:validation:MaxItems=16
+    Remove []string `json:"remove,omitempty"`
+}
+```
+
+## Examples
```

Comment:
* suggest including all implementations outlined in https://gateway-api.sigs.k8s.io/implementations/#implementations . If they dont support it, NA or skipping it should be fine